internal/resources/akscluster/resource_akscluster.go

```diff
@@ -0,0 +1,235 @@
+		if updateErr := updateClusterConfig(ctx, data, clusterResp, tc); updateErr != nil {
+			return diag.FromErr(updateErr)
+		}
+	}
+
+	// Make changes to cluster nodepools.
+	if data.HasChange("spec.0.nodepool") {
+		if npChangeErr := handleNodepoolChanges(ctx, nodepoolResp.Nodepools, data, tc.TMCConnection); npChangeErr != nil {
+			return diag.FromErr(npChangeErr)
+		}
+	}
+
+	return dataSourceTMCAKSClusterRead(ctx, data, config)
+}
+
+// resourceClusterDelete deletes an AKS cluster and all associated node pools.
+func resourceClusterDelete(ctx context.Context, data *schema.ResourceData, config any) diag.Diagnostics {
+	tc, ok := config.(authctx.TanzuContext)
+	if !ok {
+		return diag.Errorf("error while retrieving Tanzu auth config")
+	}
+
+	fn := extractClusterFullName(data)
+	if err := tc.TMCConnection.AKSClusterResourceService.AksClusterResourceServiceDelete(fn, "false"); err != nil && !clienterrors.IsNotFoundError(err) {
+		return diag.FromErr(errors.Wrapf(err, "Unable to delete Tanzu Mission Control AKS cluster entry, name : %s", data.Get(NameKey)))
+	}
+
+	ctx, cancel := context.WithTimeout(ctx, getTimeOut(data))
+	defer cancel()
+
+	if err := pollUntilClusterDeleted(ctx, data, tc.TMCConnection.AKSClusterResourceService, getPollInterval(ctx)); err != nil {
+		return diag.FromErr(err)
+	}
+
+	return diag.Diagnostics{}
+}
+
+// validate returns an error configuration will result in a cluster that will fail to create.
+func validate(nodepools []*models.VmwareTanzuManageV1alpha1AksclusterNodepoolNodepool) error {
+	for _, n := range nodepools {
+		if *n.Spec.Mode == models.VmwareTanzuManageV1alpha1AksclusterNodepoolModeSYSTEM {
+			return nil
+		}
+	}
+
+	return errors.New("AKS cluster must contain at least 1 SYSTEM nodepool")
+}
+
+// createOrUpdateCluster creates an AKS cluster in TMC.  It is possible the cluster already exists in which case the
+// existing cluster is updated with any node pools defined in the configuration.
+func createOrUpdateCluster(data *schema.ResourceData, client akscluster.ClientService) error {
+	cluster := ConstructCluster(data)
+	clusterReq := &models.VmwareTanzuManageV1alpha1AksclusterCreateAksClusterRequest{AksCluster: cluster}
+	createResp, err := client.AksClusterResourceServiceCreate(clusterReq)
+
+	if clienterrors.IsAlreadyExistsError(err) {
+		if getErr := getExistingCluster(data, client, clusterReq); getErr != nil {
+			return errors.Wrapf(getErr, "Failed to created cluster do to conflict but conflicting cluster not found")
+		}
+
+		return nil
+	}
+
+	if err != nil {
+		return err
+	}
+
+	data.SetId(createResp.AksCluster.Meta.UID)
+
+	return nil
+}
+
+func getExistingCluster(data *schema.ResourceData, client akscluster.ClientService, clusterReq *models.VmwareTanzuManageV1alpha1AksclusterCreateAksClusterRequest) error {
+	getResp, getErr := client.AksClusterResourceServiceGet(clusterReq.AksCluster.FullName)
+	if getErr != nil {
+		return getErr
+	}
+
+	data.SetId(getResp.AksCluster.Meta.UID)
```

Comment:
If the cluster exists the data is set on line L152 because we return from the function early on L147.  I can move this into that return block if you think it makes it more clear.

Reply:
Got it, existing code should be good. No need to modify